src/varity/vcf_to_hgvs/protein.clj

```diff
@@ -157,6 +157,22 @@
         alt-positions (set (range pos (inc end-pos)))]
     (boolean (seq (s/intersection ter-site-positions alt-positions)))))
 
+(defn- ref-include-from-ter-start-and-over-ter-end?
+  [{:keys [strand cds-start cds-end]} pos ref alt]
+  (let [[del _ offset _] (diff-bases ref alt)
+        pos (+ pos offset)
+        ndel (count del)
+        ter-start-pos (if (= strand :forward)
+                        (- cds-end 2)
+                        (+ cds-start 2))
+        ter-end-pos (if (= strand :forward)
+                      cds-end
+                      cds-start)
+        pos-end (+ pos (if (= ndel 0) 0 (dec ndel)))]
+    (if (= strand :forward)
+      (and (= pos ter-start-pos) (<= ter-end-pos pos-end))
+      (and (= pos-end ter-start-pos) (<= pos ter-end-pos)))))
```

Comment:
I added this fn to determine following condition.
`deletions starting 5' of and including the translation termination (stop) codon are described as frameshift.`
https://hgvs-nomenclature.org/stable/recommendations/protein/frameshift/